src/IAM/Domain/User/Model/UserRepository.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Twitter\IAM\Domain\User\Model;
+
+interface UserRepository
+{
+    public function save(User $user): void;
+
+    public function existsByEmail(Email $email): bool;
```

Comment:
```suggestion
    public function exists(Email $email): bool;
```
https://www.doctrine-project.org/projects/doctrine-collections/en/2.4/index.html#exists
Check what the Specification pattern.

Provide more tradeoffs.

Reply:
The chosen name immediately tells the developer what it does and how, clarifying its intent. Specification pattern may be great for complex comparisons, but I am confident that in our case it will lead to unnecessary complexity.

See [what AI says](https://chatgpt.com/share/69464a79-1ac8-8006-a985-5f4cd09973a7) not only about using this pattern, but also about choosing other method names.